src/main/groovy/com/bmuschko/gradle/docker/DockerRegistryCredentials.groovy

```diff
@@ -16,38 +16,47 @@
 package com.bmuschko.gradle.docker
 
 import groovy.transform.CompileStatic
-import org.gradle.api.credentials.PasswordCredentials
+import org.gradle.api.Project
+import org.gradle.api.provider.Property
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.Optional
 
 @CompileStatic
-class DockerRegistryCredentials implements PasswordCredentials {
+class DockerRegistryCredentials {
```

Comment:
Do you want to do what I suggested here and offer a `asPasswordCredentials` method on this type?
https://github.com/bmuschko/gradle-docker-plugin/pull/655#issuecomment-424041169